Mathlib/Combinatorics/SimpleGraph/Connectivity/Subgraph.lean

```diff
@@ -440,25 +443,34 @@ lemma _root_.SimpleGraph.Walk.toSubgraph_connected {u v : V} (p : G.Walk u v) :
   induction p with
   | nil => apply singletonSubgraph_connected
   | @cons _ w _ h p ih =>
-    apply (subgraphOfAdj_connected h).sup ih
-    exists w
+    refine Subgraph.connected_sup (subgraphOfAdj_connected h).preconnected ih.preconnected ⟨w, ?_⟩
     simp
 
+lemma induce_union_connected_of_nonempty_inf {H : G.Subgraph} {s t : Set V}
+    (sconn : (H.induce s).Preconnected) (tconn : (H.induce t).Preconnected)
+    (sintert : (s ⊓ t).Nonempty) :
+    (H.induce (s ∪ t)).Connected :=
+  (Subgraph.connected_sup sconn tconn sintert).mono le_induce_union <| by simp
+
+@[deprecated induce_union_connected_of_nonempty_inf (since := "2025-10-22")]
 lemma induce_union_connected {H : G.Subgraph} {s t : Set V}
     (sconn : (H.induce s).Connected) (tconn : (H.induce t).Connected)
     (sintert : (s ⊓ t).Nonempty) :
-    (H.induce (s ∪ t)).Connected := by
-  refine (sconn.sup tconn sintert).mono ?_ ?_
-  · apply le_induce_union
-  · simp
+    (H.induce (s ∪ t)).Connected :=
+  induce_union_connected_of_nonempty_inf sconn.preconnected tconn.preconnected sintert
```

Comment:
In cases like this, my suggestion is to keep the old name, and just adjust the statement. It's a breaking change, but the old name was suitable (and arguably better), and it avoids a deprecation cycle for these lemmas.